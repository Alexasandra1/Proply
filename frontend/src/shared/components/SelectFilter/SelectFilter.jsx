// SelectFilter.js
import React from 'react';
import './SelectFilter.scss';

export function SelectFilter(props) {
    const handleSelectChange = (event) => {
        props.onChange(event.target.value);
    };

    return (
        <div className={`selectFilter__block ${props.styleSelectFilter}`}>
            <select
                className={`selectFilter__block__select`}
                value={props.value}
                onChange={handleSelectChange}
            >
                <option
                    className={`selectFilter__block__select__optionName`}
                    value="0"
                >
                    {props.optionWordName}
                </option>
                <option
                    className={`selectFilter__block__select__optionFirst`}
                    value="1"
                >
                    {props.optionWord}
                </option>
                <option
                    className={`selectFilter__block__select__optionSecond`}
                    value="2"
                >
                    {props.optionWord2}
                </option>
                <option
                    className={`selectFilter__block__select__optionThird`}
                    value="3"
                >
                    {props.optionWord3}
                </option>
                <option
                    className={`selectFilter__block__select__optionFourth`}
                    value="4"
                >
                    {props.optionWord4}
                </option>
            </select>
        </div>
    );
}


// import React from 'react';
// import './SelectFilter.scss';

// export function SelectFilter(props) {
//     return (
//         <div className={`selectFilter__block ${props.styleSelectFilter}`}>
//             <select className={`selectFilter__block__select`}>
//                 <option className={`selectFilter__block__select__optionName`}
//                     value={props.value}>{props.optionWordName}</option>
//                 <option className={`selectFilter__block__select__optionFirst`}
//                     value={props.value}>{props.optionWord}</option>
//                 <option className={`selectFilter__block__select__optionSecond`}
//                     value={props.value2}>{props.optionWord2}</option>

//             </select>
//         </div>
//     );
// }

