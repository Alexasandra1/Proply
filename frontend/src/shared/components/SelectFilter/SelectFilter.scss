select {
    text-align: right;
    font-family: Montserrat;
    font-size: 32px;
    background: transparent;
    border-color: transparent;
}

// select option {
//     color: #FFF;
//     font-family: Montserrat;
//     background-color: black;
// }