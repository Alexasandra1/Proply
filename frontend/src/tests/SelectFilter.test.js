import React from 'react';
import renderer from 'react-test-renderer';
import { SelectFilter } from '../shared/components/SelectFilter/SelectFilter';

describe('SelectFilter', () => {
    it('renders correctly', () => {
        const tree = renderer
            .create(<SelectFilter 
                styleSelectFilter="style" 
                value="0" 
                optionWordName="Option 1" 
                optionWord="Option 2" 
                optionWord2="Option 3" 
            />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});
