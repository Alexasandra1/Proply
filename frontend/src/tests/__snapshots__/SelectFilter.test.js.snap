// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectFilter renders correctly 1`] = `
<div
  className="selectFilter__block style"
>
  <select
    className="selectFilter__block__select"
    onChange={[Function]}
    value="0"
  >
    <option
      className="selectFilter__block__select__optionName"
      value="0"
    >
      Option 1
    </option>
    <option
      className="selectFilter__block__select__optionFirst"
      value="1"
    >
      Option 2
    </option>
    <option
      className="selectFilter__block__select__optionSecond"
      value="2"
    >
      Option 3
    </option>
    <option
      className="selectFilter__block__select__optionThird"
      value="3"
    />
    <option
      className="selectFilter__block__select__optionFourth"
      value="4"
    />
  </select>
</div>
`;
